set.mm

```diff
@@ -54049,20 +54049,33 @@ the restriction (of the relation) to the singleton containing this
       $.
   $}
 
+  ${
+    $d x y A $.  $d x y B $.
+    $( The intersection of the identity relation with a cartesian product is
+       the restriction of the identity relation to the intersection of the
+       factors.  (Contributed by FL, 2-Aug-2009.)  (Proof shortened by Peter
+       Mazsa, 9-Sep-2022.)  Generalize statement from cartesian square (now
+       ~ idinxpresid ) to cartesian product.  (Revised by BJ, 23-Sep-2023.) $)
+    idinxpres $p |- ( _I i^i ( A X. B ) ) = ( _I |` ( A i^i B ) ) $=
```

Comment:
Nice. I have a similar result
```metamath
  ${
    $d A x $.  $d B x $.  $d R x $.
    $( Two ways to say that an intersection of the identity relation with a
       Cartesian product is a subclass.  (Contributed by Peter Mazsa,
       12-Dec-2023.) $)
    ref4 $p |- ( ( _I i^i ( A X. B ) ) C_ R <-> ( _I |` ( A i^i B ) ) C_ R ) $=
      ( vx cid cxp cin wss cv wbr wral cres ref5 idref bitr4i ) EABFGCH
    DIZPCJDABGZKEQLCHDABCMDQCNO $.
  $}
```
based on
```metamath
  ${
    $d A x y $.  $d B x y $.  $d R x y $.
    $( Two ways to say that an intersection of the identity relation with a
       Cartesian product is a subclass.  (Contributed by Peter Mazsa,
       12-Dec-2023.) $)
    ref5 $p |- ( ( _I i^i ( A X. B ) ) C_ R <-> A. x e. ( A i^i B ) x R x ) $=
      ( vy cv wceq wbr wi wral wcel cid cxp cin wss equcom imbi1i breq2
    ralbii ceqsralv2 bitr3i idinxpss ralin 3bitr4i ) AFZEFZGZUEUFDHZI
    ZECJZABJUECKUEUEDHZIZABJLBCMNDOUKABCNJUJULABUJUFUEGZUHIZECJULUNUI
    ECUMUGUHEAPQSUHUKEUECUFUEUEDRTUASAEBCDUBUKABCUCUD $.
  $}
```
but this is more general.

Reply:
Thanks. Yes, I think idinxpres subsumes ref4 (which can be proved from it using sseq1i).